framework/include/variables/MooseVariableFV.h

```diff
@@ -365,7 +414,103 @@ class MooseVariableFV : public MooseVariableField<OutputType>
     return _element_data->hasDirichletBC() || _neighbor_data->hasDirichletBC();
   }
 
+  std::pair<bool, const FVDirichletBC *> getDirichletBC(const FaceInfo & fi) const;
+
+  std::pair<bool, std::vector<const FVFluxBC *>> getFluxBCs(const FaceInfo & fi) const;
+
+  void residualSetup() override;
+  void jacobianSetup() override;
+
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  /**
+   * Get the solution value for the provided element and seed the derivative for the corresponding
+   * dof index
+   * @param elem The element to retrieive the solution value for
+   */
+  ADReal getElemValue(const Elem * elem) const;
+
+  ADReal getNeighborValue(const Elem * const neighbor,
+                          const FaceInfo & fi,
+                          const ADReal & elem_value) const;
+
+  ADReal
+  getFaceValue(const Elem * const neighbor, const FaceInfo & fi, const ADReal & elem_value) const;
+
+  const ADReal & getVertexValue(const Node & vertex) const;
+
+  /**
+   * Get custom coefficients on a per-element basis. These should correspond to \p a
+   * coefficients in the notation of Moukallad's "Finite Volume Method in Computational Fluid
+   * Dynamics"
+   */
+  const ADReal &
+  adCoeff(const Elem * elem, void * context, ADReal (*fn)(const Elem * const, void *)) const;
```

Comment:
I actually think is bad design. These coefficients are really specific to the Rhie-Chow interpolation and I put them here only because I didn't quickly think of another way to store them in a single cache of data. I'm going to strip this out for now and if I really don't think of a better way to do this I can add this back in the NSFV PR (but I really should think of a better way)